warp10/src/main/java/io/warp10/json/JsonUtils.java

```diff
@@ -0,0 +1,214 @@
+          encoded = encoder.addElement(sb, value);
+          if (encoded) {
+            break;
+          }
+        }
+        if (encoded) {
+          gen.writeRaw(sb.toString());
+        } else {
+          // No custom encoders able to encode this object, write null.
+          gen.writeNull();
+        }
+      } else {
+        // No custom encoders defined, write null.
+        gen.writeNull();
+      }
+    }
+  }
+
+  public static final NullKeySerializer nullKeySerializer = new NullKeySerializer();
+  public static final CustomEncodersSerializer customEncodersSerializer = new CustomEncodersSerializer();
+
+  //
+  // ObjectMapper instances are thread-safe, so we can safely use a single static instance.
+  //
+  private static final ObjectMapper strictMapper;
+  private static final ObjectMapper looseMapper;
+
+  public interface JsonEncoder {
+    boolean addElement(StringBuilder sb, Object o);
+  }
+
+  private static List<JsonEncoder> encoders;
+
+  static {
+    //
+    // Configure a module to handle the serialization of non-base classes.
+    //
+    SimpleModule module = new SimpleModule();
+    // Add the NotSerializedToCustomSerializedModifier instance
+    module.setSerializerModifier(new NotSerializedToCustomSerializedModifier());
+    // Add core custom serializers
+    module.addSerializer(new GeoTimeSerieSerializer());
+    module.addSerializer(new GTSEncoderSerializer());
+    module.addSerializer(new MetadataSerializer());
+    module.addSerializer(new NamedWarpScriptFunctionSerializer());
+    module.addSerializer(new MacroSerializer());
+    module.addSerializer(new BytesSerializer());
+    module.addSerializer(new RealVectorSerializer());
+    module.addSerializer(new RealMatrixSerializer());
+
+    //
+    // Common configuration for both strict and loose mappers.
+    //
+    JsonFactoryBuilder builder = new JsonFactoryBuilder();
+    builder.enable(JsonReadFeature.ALLOW_NON_NUMERIC_NUMBERS);
+    builder.enable(JsonReadFeature.ALLOW_MISSING_VALUES);
+    builder.enable(JsonWriteFeature.ESCAPE_NON_ASCII);
+    builder.disable(JsonWriteFeature.WRITE_NUMBERS_AS_STRINGS);
+    builder.disable(StreamWriteFeature.AUTO_CLOSE_TARGET);
+
+    //
+    // Configure strict mapper
+    //
+    builder.enable(JsonWriteFeature.WRITE_NAN_AS_STRINGS);
+    strictMapper = new ObjectMapper(builder.build());
+    strictMapper.getSerializerProvider().setNullKeySerializer(nullKeySerializer);
+    strictMapper.registerModule(module);
+
+    //
+    // Configure loose mapper
+    //
+    builder.disable(JsonWriteFeature.WRITE_NAN_AS_STRINGS);
+    looseMapper = new ObjectMapper(builder.build());
+    looseMapper.getSerializerProvider().setNullKeySerializer(nullKeySerializer);
+    looseMapper.registerModule(module);
+  }
+
+  //
+  // Method to deserialize JSON to Objects.
+  //
+
+  public static Object jsonToObject(String json) throws JsonProcessingException {
+    return strictMapper.readValue(json, Object.class);
```

Comment:
Are integer numbers correctly converted to LONGs by this?

Reply:
Yes, and floating-point numbers to Double.